Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchyBasedLinkResolver.swift

```diff
@@ -58,6 +57,42 @@ final class PathHierarchyBasedLinkResolver {
         }
     }
     
+    /// Returns the direct descendants of the given page that match the given source language filter.
+    ///
+    /// A descendant it included if it has a language representation in at least one of the languages in the given language filter.
+    ///
+    /// - Parameters:
+    ///   - reference: The identifier of the page whose descendants to return.
+    ///   - languagesFilter: A set of source languages to filter descendants against.
+    /// - Returns: The references of each direct descendant that has a language representation in at least one of the given languages.
+    func directDescendants(of reference: ResolvedTopicReference, languagesFilter: Set<SourceLanguage>) -> Set<ResolvedTopicReference> {
+        guard let id = resolvedReferenceMap[reference] else { return [] }
+        let node = pathHierarchy.lookup[id]!
+        
+        func directDescendants(of node: PathHierarchy.Node) -> [ResolvedTopicReference] {
+            return node.children.flatMap { _, container in
+                container.storage.compactMap { element in
+                    guard let childID = element.node.identifier, // Don't include sparse nodes
+                          !element.node.isDisfavoredInCollision, // Don't include disfavored collisions
```

Comment:
The topic graph implementation [filters out all default implementations](https://github.com/apple/swift-docc/pull/854/files/74f5baaa74c8f56f0da668a805f513da46d736e1..5c889232ff055adc9a6020110b09607b20ab0378#diff-72a1201a198063334dfa099006282da39631d1eaebcb3b1b3ecb0589dadd66b1R65-R73). This is essentially avoiding adding a reference that's known to be immediately filtered out.

Reply:
What about future work like #846, which [sets nodes as disfavored when they're part of an overload group](https://github.com/apple/swift-docc/pull/846/files#diff-e0d0f88acc1bca134d47867a977e60a988a20118c4b52451f90376f6e3d56f7cR123-R130)?

Reply:
That depends on if we'd want automatic curation to list the individual overloads or just the overload group? IMO listing only the overload group feels inline with the overall goal of that overload group but if we want to change this in the future it would be easy to do so.

Reply:
The ultimate goal is for automatic curation to only list the overload group, but i was expecting it to be covered in future work. 😅 If we're going to make this a rule of automatic curation, though, it might be worth adding a comment somewhere, whether on the isDisfavored property or here on the automatic curation method.

Reply:
I could also change isDisfavored property from a Bool to an enum/option set with different behaviors if we want more control over where and how the node is disfavored

Reply:
That could work! That way it's immediately apparent whether we want something to still appear in automatic curation but make it not interfere in link resolution.

Reply:
Fixed in https://github.com/apple/swift-docc/pull/854/commits/22563c98f733b3ac2f31023f6e6d8262a5305cc1